### DISCUSSION TITLE
Is the connection pooler / supavisor broken for anyone else?

### DISCUSSION THREAD

**Discussion by haydn** [supabase/supabase#26879] [Questions]
We've had to connect our app directly to the Postgres database and bypass the connection pooler altogether because of persistent errors it appears to be causing (and we've had to pay for the IPv4 add-on to do that).

Here's a graph of the errors we were seeing and where it stops being a problem on May 29 is where we switched from the pooler connection to the direct connection:

<img width="1088" alt="image" src="https://github.com/supabase/supabase/assets/27211/6cc1766e-8faa-4ab4-bd5a-1d3a86e47347">

The error we were seeing in the pooler logs was "Terminating with reason" as I've describing in this bug report:

https://github.com/supabase/supavisor/issues/344

That was manifesting in our app as "Connection terminated unexpectedly" errors being emitted by node-postgres.

Is anyone else seeing anything like this?

~~I've raised a support request with Supabase about it, but it was forwarded to the "pooler team" 3 days ago and I haven't heard anything since then. 🤷~~

Response from the pooler team:

This log means that the client has initiated a disconnect, but the linked direct connection was not in an idle state.

**Comment by dedlockdave** [comment]
thanks for creating this issue . been a pain in the ass for me, but ill try to connect directly as you did.
